app/reducers/articleReducers.js

```diff
@@ -0,0 +1,34 @@
+import { ADD_HIGHLIGHT, TEST_ACTION } from '../actions/actionTypes';
+
+// Note: not 100% sure this is the 'proper' reducer layout - we'll find out more
+// as we go
+
+const initialState = {
+  exampleProp: 'defaultValue',
+  highlights: []
+}
+
+export default function articleReducer(state = initialState, action) {
+  switch (action.type) {
+    case TEST_ACTION:
+      return Object.assign({}, state, { exampleProp: action.text });
+    case ADD_HIGHLIGHT:
```

Comment:
feels to me like business logic should be defined in the action and it should come ready into the reducer.